cmd/update-notifier_test.go

```diff
@@ -17,34 +17,46 @@
 package cmd
 
 import (
-	"runtime"
 	"strings"
 	"testing"
 	"time"
-
-	"github.com/fatih/color"
 )
 
 // Tests update notifier string builder.
-func TestUpdateNotifier(t *testing.T) {
+func TestComputeUpdateMessage(t *testing.T) {
 	plainMsg := "You are running an older version of Minio released "
-	colorMsg := plainMsg
-	yellow := color.New(color.FgYellow, color.Bold).SprintfFunc()
-	if runtime.GOOS == "windows" {
-		plainMsg += "3 days from now"
-		colorMsg += yellow("3 days from now")
-	} else {
-		plainMsg += "2 days from now"
-		colorMsg += yellow("2 days from now")
-	}
 
-	updateMsg := colorizeUpdateMessage(minioReleaseURL, time.Duration(72*time.Hour))
+	testCases := []struct {
+		older time.Duration
+		dlURL string
 
-	if !(strings.Contains(updateMsg, plainMsg) || strings.Contains(updateMsg, colorMsg)) {
-		t.Fatal("Duration string not found in colorized update message", updateMsg)
+		expectedSubStr string
+	}{
+		{72 * time.Hour, "my_download_url", "days"},
+		{3 * time.Hour, "https://my_download_url_is_huge/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "hours"},
+		{-72 * time.Hour, "another_update_url", ""},
+		{0, "another_update_url", ""},
```

Comment:
Ok done. The humanize lib is not that smart. It always says the same thing. Also with 1 second the rel time func prints "now" - I guess this is because of some rounding issue. It is not a problem.